app_dart/test/request_handlers/get_benchmarks_test.dart

```diff
@@ -92,6 +96,7 @@ void main() {
     });
 
     test('reports statuses with input branch', () async {
+      config.maxRecordsValue = 2;
```

Comment:
Do we need to add tests that are adding records from master and one release branch?

Reply:
`reports statuses with input branch` is testing this case.

Reply:
In that case can we refactor the tests to cover more atomic cases? Having tests that are testing too much logic require to understand the entire workflow to fix them when they  break because we don't know exactly what is failing.

Reply:
could you please elaborate on what you mean by covering atomic cases. Do you mean test their sub functions individually? In that case, we may need to move those private functions public.

BTW: Each API in cocoon performs a specific purpose. It seems that all APIs' tests test their whole logic no matter the API is as complex as https://github.com/flutter/cocoon/blob/master/app_dart/lib/src/request_handlers/check_for_waiting_pull_requests.dart, or as simply as https://github.com/flutter/cocoon/blob/master/app_dart/lib/src/request_handlers/get_branches.dart.

To make the test more easily readable, I have added some documents.

Reply:
Sent the link to tests best practices by email, please read it and refactor accordingly.

Reply:
Thanks for the link! Have added more atomic tests.